felix/calc/l3_route_resolver.go

```diff
@@ -930,13 +1027,27 @@ func (r *RouteTrie) RemoveRef(cidr ip.V4CIDR, nodename string, rt RefType) {
 	})
 }
 
-func (r *RouteTrie) SetRouteSent(cidr ip.V4CIDR, sent bool) {
+func (r *RouteTrie) SetRouteSent(cidr ip.CIDR, sent bool) {
 	r.updateCIDR(cidr, func(ri *RouteInfo) {
 		ri.WasSent = sent
 	})
 }
 
-func (r RouteTrie) updateCIDR(cidr ip.V4CIDR, updateFn func(info *RouteInfo)) bool {
+func (r RouteTrie) updateCIDR(cidr ip.CIDR, updateFn func(info *RouteInfo)) bool {
+	if cidr == nil {
+		logrus.WithField("cidr", cidr).Debug("Ignoring nil CIDR update")
+		return false
+	}
+
+	var trie *ip.CIDRTrie
+	switch cidr.(type) {
```

Comment:
Looks like we have this switch pattern scattered throughout - might be nice to clean it up with a helper.

```go
trie := r.trieForCIDR(cidr)
```

Alternatively, we could remove the address family special casing from here and instead handle it up a level in the calling code.

e.g., instead of the RouteTrie having both a `v4T` and `v6T`, it can just have a `trie` and then the calling code instantiantes two `RouteTrie` objects.

^ This is more in line with how I was originally imagining this code, but isn't worth refactoring the whole thing over unless you think its simpler (might not be, I haven't had my head in this code for some time).